### DISCUSSION TITLE
Using the default description with full match body in TheHive alerter

### DISCUSSION THREAD

**Discussion by DrewHorrigan** [jertel/elastalert2#1136] [Q&A] [answered]
Hi. I am currently using TheHive alerter, and my intent is to make sure TheHive alerts include the rule's match body as is. This is done by default if `alert_text` and TheHive `description` is not specified, but there is a problem with it - it's submitted to TheHive in plain text, and TheHive expects Markdown content, so the alert description is displayed without line breaks, making it mostly unreadable.

I've been trying to solve this by surrounding the description with monospace formatting characters (\`\`\`), but in this case I don't know how to include _all_ fields in the description, as I can only include specific fields using Python formatting and `description_args`. Here are several things I've tried:

- Using fields from documentation that could include all key-value pairs from the alert (using names from ['Alert Content' section](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#alert-content)), such as `field_values`, in `description_args` - these seem to be placeholder names rather than actual addressable fields, and result in empty description.
- Using Jinja templating for alert_text - in this case I managed to iterate through `_data` field, but I found that this also displays a lot of internal information, such as rule-specific fields, which I do not want to include. This also has the drawback of using the same text on other alerters.
- When iterating through Jinja's `_data` object, I saw that it has other embedded Python objects - `rule.type` and `rule.alert`, but I could not extract the match_body from either of them (`_data.type.__dict__.matches` is an empty array).
- I saw that Markdown is supported for the summary table, but this is also not what I want.

With all that in mind, is there a way to include the full alert content in TheHive in readable form without editing the source code?

**Comment by jertel** [comment, marked as answer]
I think the proper solution is to submit a PR that provides a toggle flag that allows rules to opt-in to default markdown formatting, so that future releases will have the improvement. My guess is that most people want specific fields in the alert, not the entire document. So if you want the changed behavior you may need to submit the PR yourself.

**Reply by DrewHorrigan** [reply]
Okay then, thanks. :(
